File: AdminArea/AdminArea.Web/Modules/souqcom/Cartcom/CartcomColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminArea.souqcom.Columns;

[ColumnsScript("souqcom.Cartcom")]
[BasedOnRow(typeof(CartcomRow), CheckNames = true)]
public class CartcomColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public int UserId { get; set; }
    public string ProductName { get; set; }
    public int Qty { get; set; }
}

File: AdminArea/AdminArea.Web/Modules/ServerTypes/souqcom/CartcomColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CartcomRow } from "./CartcomRow";

export interface CartcomColumns {
    Id: Column<CartcomRow>;
    UserId: Column<CartcomRow>;
    ProductName: Column<CartcomRow>;
    Qty: Column<CartcomRow>;
}

export class CartcomColumns extends ColumnsBase<CartcomRow> {
    static readonly columnsKey = 'souqcom.Cartcom';
    static readonly Fields = fieldsProxy<CartcomColumns>();
}